### PR TITLE
Adding DoWhy python package

### PR DESCRIPTION
**Pull request by amit-sharma**
Hello, 

This is a PR for adding my package DoWhy to conda-forge.
The package has MIT license and it is already included in the tar archive's root folder. Therefore, following the instructions [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#about), I have added the path to my LICENSE file in meta.yaml.

Let me know if there are any further changes that need to be made. Thanks!

@conda-forge-admin, please ping team

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/dowhy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - pip
+    - python
+  run:
+    - econml >=0.5
+    - matplotlib
+    - networkx >=2.0
+    - numpy >=1.15
+    - pandas >=0.24
+    - pydot >=1.4
+    - python
```
Comment (chrisburr) [inline recipes/dowhy/meta.yaml]: If (and only if) it's noarch you can constrain Python using:
```suggestion
    - python >=3.4
```

File: recipes/dowhy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  skip: True  # [py<34]
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (chrisburr) [inline recipes/dowhy/meta.yaml]: If (and only if) it's noarch you can constrain Python using:
```suggestion
    - python >=3.4
```

File: recipes/dowhy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 99d58f0098cf37f2b69073f9b47253cafe13f1136d694cd972bec98e9c6ee484
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  skip: True  # [py<34]
```
Comment (chrisburr) [inline recipes/dowhy/meta.yaml]:
```suggestion
  noarch: python
```

File: recipes/dowhy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - dowhy.do_samplers
+
+about:
+  home: "https://github.com/microsoft/dowhy"
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE  # LICENSE file is in the root folder of the archive
+  summary: "DoWhy is a Python library for causal inference that supports explicit modeling and testing of causal assumptions."
+  doc_url: "https://microsoft.github.io/dowhy/" 
+  dev_url:
```
Comment (chrisburr) [inline recipes/dowhy/meta.yaml]: If there isn't a URL this field should be removed entirely:
```suggestion
```

File: recipes/dowhy/meta.yaml (hunk 2 above)
Comment (amit-sharma) [reply recipes/dowhy/meta.yaml]: thank you @chrisburr for your super-quick response! I have added all your suggested changes.

**Comment by amit-sharma** [comment]
Apologies--didn't realize that one of the dependencies is not on conda. Let me contact the developers of the dependency package--allow me some time.

**Comment by chrisburr** [comment]
No problem! You can also add it yourself if you'd prefer (assuming it has a reasonable licence).

**Comment by amit-sharma** [comment]
Adding an updated PR for dowhy's latest version.

**Pull request closed without merging**